truss-train/truss_train/framework.py

```diff
@@ -0,0 +1,36 @@
+import contextlib
+import importlib.util
+import os
+import pathlib
+from typing import Iterator
+
+from truss_train import definitions
+
+
+@contextlib.contextmanager
```

Comment:
I know we just went over this in the offsite, but I actually think this code is different enough from [chains](https://github.com/basetenlabs/truss/blob/536a1f7af1f04b7fd93814d7f69b566dd89aaac1/truss-chains/truss_chains/framework.py?plain=1#L1264-L1271) to warrant the duplication, thoughts? Differences include:
- no module modifications needed
- targets aren't subclasses, but actual instances of `TrainingProject`
- error messages